python/cudf/cudf/core/column/column.py

```diff
@@ -70,44 +71,34 @@ def as_frame(self, name=None):
     @property
     def data_array_view(self):
         """
-        View the data as a device array or nvstrings object
+        View the data as a CuPy ndarray or nvstrings object
         """
         if self.dtype == "object":
             return self.nvstrings
 
         if is_categorical_dtype(self.dtype):
             return self.codes.data_array_view
-        else:
-            dtype = self.dtype
 
-        result = rmm.device_array_from_ptr(
-            ptr=self.data.ptr, nelem=len(self), dtype=dtype
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.data).view(self.dtype)
         return result
 
     @property
     def mask_array_view(self):
         """
-        View the mask as a device array
+        View the mask as a CuPy ndarray
         """
-        result = rmm.device_array_from_ptr(
-            ptr=self.mask.ptr,
-            nelem=calc_chunk_size(len(self), mask_bitsize),
```

Comment:
Seems like we still need to handle this somehow. Maybe we can add an `is_bits` attribute to `Buffer` to check whether this kind of handling is needed?

Reply:
No, Buffer shouldn't be aware of whether it's a bitmask or not. This actually shouldn't be an issue as the Buffer should be sized properly already when it was created.

Reply:
Went ahead and pulled this change out as PR ( https://github.com/rapidsai/cudf/pull/4079 ) just to make sure I'm not missing anything 🙂